libres/tests/analysis/test_compare_initX.cpp

```diff
@@ -0,0 +1,181 @@
+
+            std_enkf_initX(std_enkf_data, X1, nullptr, S, R, nullptr, E, D,
+                           rng);
+            ies_enkf_initX(std_enkf_get_truncation(std_enkf_data),
+                           std_enkf_get_subspace_dimension(std_enkf_data),
+                           std_enkf_data_get_inversion(std_enkf_data), S, R, E,
+                           D, X2);
+
+            REQUIRE(matrix_similar(X1, X2, 1e-10));
+            std_enkf_data_free(std_enkf_data);
+            matrix_free(X2);
+            matrix_free(X1);
+        }
+
+        WHEN("The inversion_subspace_EE_R method is used to calculate X") {
+            std_enkf_data_type *std_enkf_data =
+                static_cast<std_enkf_data_type *>(std_enkf_data_alloc());
+
+            matrix_type *X1 = matrix_alloc(ens_size, ens_size);
+            matrix_type *X2 = matrix_alloc(ens_size, ens_size);
+            std_enkf_set_string(std_enkf_data, INVERSION_KEY,
+                                STRING_INVERSION_SUBSPACE_EE_R);
+
+            std_enkf_initX(std_enkf_data, X1, nullptr, S, R, nullptr, E, D,
+                           rng);
+            ies_enkf_initX(std_enkf_get_truncation(std_enkf_data),
+                           std_enkf_get_subspace_dimension(std_enkf_data),
+                           std_enkf_data_get_inversion(std_enkf_data), S, R, E,
+                           D, X2);
+
+            REQUIRE(matrix_similar(X1, X2, 1e-10));
+            std_enkf_data_free(std_enkf_data);
+            matrix_free(X2);
+            matrix_free(X1);
+        }
+
+        WHEN("The inversion method inversion_EXACT_R is used to callulate X") {
+            std_enkf_data_type *std_enkf_data =
+                static_cast<std_enkf_data_type *>(std_enkf_data_alloc());
+
+            matrix_type *X1 = matrix_alloc(ens_size, ens_size);
+            matrix_type *X2 = matrix_alloc(ens_size, ens_size);
+            std_enkf_set_string(std_enkf_data, INVERSION_KEY,
+                                STRING_INVERSION_SUBSPACE_EXACT_R);
+
+            std_enkf_initX(std_enkf_data, X1, nullptr, S, R, nullptr, E, D,
+                           rng);
+            ies_enkf_initX(std_enkf_get_truncation(std_enkf_data),
+                           std_enkf_get_subspace_dimension(std_enkf_data),
+                           std_enkf_data_get_inversion(std_enkf_data), S, R, E,
+                           D, X2);
+
+            REQUIRE(matrix_similar(X1, X2, 1e-10));
+            std_enkf_data_free(std_enkf_data);
+            matrix_free(X2);
+            matrix_free(X1);
+        }
+
+        WHEN("The inversion_SUBSPACE_RE method is used to calculate X") {
+            std_enkf_data_type *std_enkf_data =
+                static_cast<std_enkf_data_type *>(std_enkf_data_alloc());
+
+            matrix_type *X1 = matrix_alloc(ens_size, ens_size);
+            matrix_type *X2 = matrix_alloc(ens_size, ens_size);
+            std_enkf_set_string(std_enkf_data, INVERSION_KEY,
```

Comment:
Should probably use `GENERATE` here as well: `std_enkf_set_string(std_enkf_data, INVERSION_KEY, GENERATE(STRING_INVERSION_SUBSPACE_RE, STRING_INVERSION_SUBSPACE_EXACT_R, STRING_INVERSION_SUBSPACE_EE_R))`, unless there is some difference that I am not catching. You can also use it for `std_enkf_set_double(std_enkf_data, ENKF_TRUNCATION_KEY_, GENERATE(0.90, 0.95, 0.75,...));` or if necessary
```cpp
if(GENERATE(true, false)){
    std_enkf_set_double(std_enkf_data, 0.9);
}
```